iree/compiler/Dialect/LinalgExt/IR/test/invalid.mlir

```diff
@@ -27,11 +27,11 @@ func @sort_without_dimension(%arg0: tensor<3x4xi32>) -> tensor<3x4xi32> {
 // -----
 
 func @scatter_mixed_tensor_memref(
-    %update : memref<?x?xf32>, %indices : tensor<?xi32>,
+    %update : memref<?x?xf32>, %indices : tensor<?x1xi32>,
     %original : tensor<?x?xf32>) -> tensor<?x?xf32> {
   // expected-error @+1 {{expected `ins` operand #0 to be of RankedTensorType}}
   %0 = linalg_ext.scatter
-      ins(%update, %indices : memref<?x?xf32>, tensor<?xi32>)
+      ins(%update, %indices : memref<?x?xf32>, tensor<?x1xi32>)
```

Comment:
I am not sure why this is `tensor<?x1xi32>`. That is same as `tensor<?xi32>`. Maybe this should be canonicalized before being converted to the `linalg_ext.scatter` op

Reply:
As offline discussion, they are technically the same. Both are fine. Following the definition from tensorflow scatter op could be easier, like no need to determine if this is `index_detph=1` case.

https://www.tensorflow.org/api_docs/python/tf/tensor_scatter_nd_update